### PR TITLE
DEPR: Remove deprecated Timestamp.freq

### PR DESCRIPTION
**Pull request by jbrockmendel**
- [ ] closes #xxxx (Replace xxxx with the GitHub issue number)
- [ ] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [ ] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [ ] Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.
- [ ] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.

### CODE REVIEW

**Comment by Dr-Irv** [comment]
Small nit here -  you are removing `freq`, not deprecating it, because it was already deprecated, right?  Might want to change the name of the PR??

**Comment by jbrockmendel** [comment]
> Might want to change the name of the PR??

This is the pattern we've stumbled into for deprecation-enforcement PRs.

**Comment by Dr-Irv** [comment]
> > Might want to change the name of the PR??
> 
> This is the pattern we've stumbled into for deprecation-enforcement PRs.

For my own education, what's the pattern for a PR that deprecates as opposed to removal?

**Comment by jbrockmendel** [comment]
> For my own education, what's the pattern for a PR that deprecates as opposed to removal?

The patterns are not distinct.  I'm open to suggestions.

**Comment by Dr-Irv** [comment]
> > For my own education, what's the pattern for a PR that deprecates as opposed to removal?
> 
> The patterns are not distinct. I'm open to suggestions.

Maybe use "DEPR:  Deprecated foobar" when deprecating, and "DEPR: Remove deprecated foobar" to correspond to the deprecation and removal of the deprecated behavior of foobar, respectively.

**Comment by mroeschke** [comment]
Updated the PR title. Also lately we have had a unofficial moratorium on merging deprecating functionality PRs until we've cleared the 1.x deprecations since the checklist issue is unfortunately not comprehensive.

File: doc/source/whatsnew/v2.0.0.rst [hunk 1]
```diff
@@ -165,6 +165,7 @@ Deprecations
 
 Removal of prior version deprecations/changes
 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+- Removed deprecated :attr:`Timestamp.freq` (:issue:`14146`)
```
Comment (mroeschke) [inline doc/source/whatsnew/v2.0.0.rst:168]: Might be good to add `freqstr` attr was removed too, and that `freq` was removed from the `Timestamp` constructor and `Timestamp.fromordinal`

**Review by mroeschke — APPROVED** [review]

**Comment by mroeschke** [comment]
Thanks @jbrockmendel

**Pull request merged**